### DISCUSSION TITLE
How to Handle Insertion into auth.users and Foreign Keys in a Custom Table with Supabase?

### DISCUSSION THREAD

**Discussion by hounaine** [supabase/supabase#30750] [Questions] [answered]
Hi,

I have a custom users table with a foreign key id referencing auth.users(id) in Supabase. During user registration, I aim to:

Create a user in auth.users using the Supabase Auth API.
Insert additional user data (e.g., bio, username, etc.) into my custom users table, using the id generated in auth.users.
However, I encounter a foreign key constraint error (fk_user_id) when trying to insert data into my users table.

Is there a better approach to handle this workflow while adhering to Supabase's design?
Should I always insert into auth.users via the Auth API, or can this be handled more effectively using a PostgreSQL stored procedure?

Thank you...

**Comment by GaryAustin1** [comment, marked as answer]
Not clear how you are doing this now.  You should be able to populate a public table and have the foreign key value point to an existing one in the auth.users table and work.

But...
Normally people use the user management guide which shows an auth.users insert trigger to populate a profile table (including the fk to auth.users as the id for that table.
https://supabase.com/docs/guides/auth/managing-user-data?queryGroups=language&language=js

**Reply by hounaine** [reply]
Got it.

**Reply by hounaine** [reply]
That was chatgpt fault not mine :)
